test/integration/shared/host_orchestration_stubs.rb

```diff
@@ -0,0 +1,19 @@
+module HostOrchestrationStubs
+  extend ActiveSupport::Concern
+
+  def click_on_submit
+    click_button('Submit')
+    wait_for_orchestration_requests # host-progress bar should disappear
+    wait_for_ajax # wait for runtime/nics/etc.. ajax requests in Host#show
+    assert page.has_link?('Edit')
+  end
+
+  def wait_for_orchestration_requests
+    Timeout.timeout(Capybara.default_max_wait_time * 2) do
+      # Notice host-progress won't be visible without a multi-threaded server
+      loop until has_no_selector?("#host-progress", :visible => false, :wait => 0.1)
+    end
+  rescue Capybara::ElementNotFound
```

Comment:
Hmm from what I'm reading that happens if the element is set to `display:none`. The default class for #host-progress (hide) is equivalent to that. I'll simplify this a bit though, I don't think the Timeout and rescue are needed.